tests/Generator/NamedArgumentsResolverTest.php

```diff
@@ -0,0 +1,304 @@
+                'foo2' => 'value 2',
+            ],
+        ];
+
+        yield 'static factory: mix of anonymous and named arguments in wrong order' => [
+            DummyWithMethods::class,
+            'create',
+            [
+                'foo2' => 'value 2',
+                'value 1',
+            ],
+            [
+                'value 1',
+                'foo2' => 'value 2',
+            ],
+        ];
+
+        yield 'method call: no named arguments' => [
+            DummyWithMethods::class,
+            'bar',
+            [
+                'value 1',
+                'value 2',
+            ],
+            [
+                'value 1',
+                'value 2',
+            ],
+        ];
+
+        yield 'method call: named arguments' => [
+            DummyWithMethods::class,
+            'bar',
+            [
+                'bar1' => 'value 1',
+                'bar2' => 'value 2',
+            ],
+            [
+                'bar1' => 'value 1',
+                'bar2' => 'value 2',
+            ],
+        ];
+
+        yield 'method call: named arguments in wrong order' => [
+            DummyWithMethods::class,
+            'bar',
+            [
+                'bar2' => 'value 2',
+                'bar1' => 'value 1',
+            ],
+            [
+                'bar1' => 'value 1',
+                'bar2' => 'value 2',
+            ],
+        ];
+
+        yield 'method call: mix of anonymous and named arguments' => [
+            DummyWithMethods::class,
+            'bar',
+            [
+                'value 1',
+                'bar2' => 'value 2',
+            ],
+            [
+                'value 1',
+                'bar2' => 'value 2',
+            ],
+        ];
+
+        yield 'method call: mix of anonymous and named arguments in wrong order' => [
+            DummyWithMethods::class,
+            'bar',
+            [
+                'bar2' => 'value 2',
+                'value 1',
+            ],
+            [
+                'value 1',
+                'bar2' => 'value 2',
+            ],
+        ];
+
+        yield 'with variadic argument' => [
+            DummyWithMethods::class,
+            'methodWithVariadic',
+            [
+                'baz2' => 'value 2',
+                'value 1',
+                'value 4',
+                'baz3' => 'value 3',
+                'value 5',
+            ],
+            [
+                'value 1',
+                'baz2' => 'value 2',
+                'value 4',
+                'value 3',
+                'value 5',
+            ],
+        ];
+
+        yield 'with missing arguments that have default values' => [
+            DummyWithMethods::class,
+            'methodWithDefaultValues',
+            [
+                'baz2' => 'value 2',
+            ],
+            [
+                'value 1',
+                'baz2' => 'value 2',
+            ],
+        ];
+
+        yield 'with method that does not exist' => [
```

Comment:
AFAICT it's also missing the case where you are passing the numeric keys as strings e.g.`'0'` which is not uncommon if you need a flag e.g. `0 (unique)`